src/components/Search/Search.tsx

```diff
@@ -1,23 +1,51 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import SearchIcon from '@material-ui/icons/Search';
 import PropTypes from 'prop-types';
+import { SearchHint } from '../SearchHint/SearchHint';
 import './Search.scss';
 
 interface SearchProps {
   testid?: string;
+  hintsFunction?: (set: React.Dispatch<React.SetStateAction<string[]>>, value: string) => void;
 }
 
-export const Search: React.FC<SearchProps> = ({ testid }) => (
-  <div className="search" data-testid={testid}>
-    <SearchIcon className="search__icon" />
-    <input data-testid={`${testid}Input`} placeholder="Search..." className="search__input" type="text" />
-  </div>
-);
+export const Search: React.FC<SearchProps> = ({ testid, hintsFunction }) => {
+  const [searchInput, setSearchInput] = useState('');
+  const [searchHints, setSearchHints] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (hintsFunction) {
+      hintsFunction(setSearchHints, searchInput);
+    }
+  }, [searchInput]);
+
+  return (
+    <div className="search" data-testid={testid}>
+      <div className="search__container">
+        <SearchIcon className="search__icon" />
+        <input
+          value={searchInput}
+          onChange={e => setSearchInput(e.target.value)}
```

Comment:
```suggestion
          onChange={event => setSearchInput(e.target.value)}
```

Reply:
same "e" is a bad practice?

Reply:
I think it's better to use descriptive consts